linen_examples/pixelcnn/model_test.py

```diff
@@ -0,0 +1,142 @@
+# Copyright 2020 The Flax Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Lint as: python3
+"""Tests for model."""
+
+from absl.testing import absltest
+from absl.testing import parameterized
+
+import numpy.testing as onp_testing
+
+from jax import random
+import jax.numpy as np
+
+from flax import nn
+
+import pixelcnn
+
+
+class ModelTest(absltest.TestCase):
+
+  def test_conv(self):
+    rng = random.PRNGKey(0)
+    x = np.arange(24).reshape(1, 4, 3, 2)
+    conv_module = pixelcnn.Conv.partial(features=4, kernel_size=(3, 2))
```

Comment:
I've updated the tests, but now two of them are failing:
```
FAILED linen_examples/pixelcnn/model_test.py::ModelTest::test_conv_down_right - AssertionError: 
FAILED linen_examples/pixelcnn/model_test.py::ModelTest::test_conv_transpose_down - AssertionError: 
```

The problem is that the mean and variance are no longer resp. 0 and 1. Example of an assertion error:
```
>     onp_testing.assert_allclose(np.mean(out, (0, 1, 2)), 0., atol=1e-5)
E     AssertionError: 
E     Not equal to tolerance rtol=1e-07, atol=1e-05
E     
E     Mismatched elements: 4 / 4 (100%)
E     Max absolute difference: 0.15736276
E     Max relative difference: inf
E      x: array([ 0.021286, -0.157363, -0.045011,  0.001434], dtype=float32)
E      y: array(0.)
```

Perhaps this is because the logic is a bit different now: in the old API we only init the model and we get an output, but in the new API we first init it with zeros, and then apply it with the input data. Do you think this may be the problem?